### PR TITLE
chore(licenseinfo): Reporting Improvements

### PR DESCRIPTION
**Pull request by dreh23**
#380

### CODE REVIEW

**Comment by maxhbr** [comment]
Please reformat the commit messages according to https://github.com/eclipse/sw360/wiki/Dev-Semantic-Commits

File: libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift [hunk 1]
```diff
@@ -101,4 +101,15 @@ service LicenseInfoService {
      */
     string getDefaultLicenseInfoHeaderText();
 
+    /**
+     * get default clearing summary text
+     */
+    string getDefaultObligationsText();
+
+    /*string getClearingSummary();
```
Comment (maxhbr) [inline libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift]: commented code

File: backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java [hunk 2]
```diff
@@ -302,8 +320,14 @@ protected LicenseInfoParsingResult noSourceParsingResult(String message) {
     }
 
     private static String loadDefaultLicenseInfoHeaderText(){
-            String defaultLicenseInfoHeader = new String( CommonUtils.loadResource(LicenseInfoHandler.class, DEFAULT_LICENSE_INFO_HEADER_FILE).orElse(new byte[0]) );
-            defaultLicenseInfoHeader = defaultLicenseInfoHeader.replaceAll("(?m)^#.*\\n", "");  // ignore comments in template file
-            return defaultLicenseInfoHeader;
+        String defaultLicenseInfoHeader = new String( CommonUtils.loadResource(LicenseInfoHandler.class, DEFAULT_LICENSE_INFO_HEADER_FILE).orElse(new byte[0]) );
+        defaultLicenseInfoHeader = defaultLicenseInfoHeader.replaceAll("(?m)^#.*\\n", "");  // ignore comments in template file
```
Comment (maxhbr) [inline backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java]: maybe extract into a function `dropCommentedLines`

File: libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift (hunk 1 above)
Comment (dreh23) [reply libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift]: Solved.

**Comment by maierthomas** [comment]
thank you for this contribution.
Please resolve the conflicts and rebase your branch to the current master.
You can use the flag `-i` (interactive) to squash your commits.

**Comment by dreh23** [comment]
#375
#377
#378
#379
#380

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayObligations.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        this.defaultText = defaultText;
+    }
+
+
+    public void doTag() throws JspException, IOException {
+        String output = project.isSetObligationsText() ? StringEscapeUtils.escapeHtml(project.obligationsText) : StringEscapeUtils.escapeHtml(defaultText);
+
+        getJspContext().getOut().print(output);
+    }
+}
```
Comment (maierthomas) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayObligations.java]: Line break

**Comment by mcjaeger** [comment]
What I did:
* installed the branch on a vagrant instance
* tested on some projects
* installed an entirely new vagrant instance with this branch
* tested a few files and few projects

-> all end up with a null pointer exception when I select "License Report as DOCX", all other tree options work. I guess there is some fundamental issue here

Stacktrace: 
```
	at org.eclipse.sw360.licenseinfo.outputGenerators.DocxGenerator.generateOutputFile(DocxGenerator.java:42)
	at org.eclipse.sw360.licenseinfo.LicenseInfoHandler.getLicenseInfoFile(LicenseInfoHandler.java:132)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Processor$getLicenseInfoFile.getResult(LicenseInfoService.java:540)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Processor$getLicenseInfoFile.getResult(LicenseInfoService.java:525)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:39)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
Dec 07, 2018 7:38:55 AM org.apache.catalina.core.StandardWrapperValve invoke
SEVERE: Servlet.service() for servlet [LicenseInfoService] in context with path [/licenseinfo] threw exception [java.lang.NullPointerException] with root cause
java.lang.NullPointerException
	at org.eclipse.sw360.licenseinfo.outputGenerators.DocxGenerator.fillOverview3rdPartyComponentTable(DocxGenerator.java:298)
	at org.eclipse.sw360.licenseinfo.outputGenerators.DocxGenerator.fillReportDocument(DocxGenerator.java:150)
	at org.eclipse.sw360.licenseinfo.outputGenerators.DocxGenerator.generateOutputFile(DocxGenerator.java:85)
	at org.eclipse.sw360.licenseinfo.outputGenerators.DocxGenerator.generateOutputFile(DocxGenerator.java:42)
	at org.eclipse.sw360.licenseinfo.LicenseInfoHandler.getLicenseInfoFile(LicenseInfoHandler.java:132)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Processor$getLicenseInfoFile.getResult(LicenseInfoService.java:540)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Processor$getLicenseInfoFile.getResult(LicenseInfoService.java:525)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:39)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
	at org.apache.thrift.server.TServlet.doPost(TServlet.java:83)
	at org.eclipse.sw360.projects.Sw360ThriftServlet.doPost(Sw360ThriftServlet.java:42)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:650)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116)
	at sun.reflect.GeneratedMethodAccessor1077.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.liferay.portal.kernel.bean.ClassLoaderBeanHandler.invoke(ClassLoaderBeanHandler.java:67)
	at com.sun.proxy.$Proxy774.doFilter(Unknown Source)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:165)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:96)
	at com.liferay.portal.kernel.servlet.PortalClassLoaderFilter.doFilter(PortalClassLoaderFilter.java:74)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:204)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:109)
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter.doFilter(InvokerFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)

2018-12-07 07:38:55,201 ERROR ProjectPortlet:283 - Error getting LicenseInfo file for project with id a076c23e7e3e6d856428e1b547009662 and generator DocxGenerator::REPORT
 org.apache.thrift.transport.TTransportException: HTTP Response code: 500
	at org.apache.thrift.transport.THttpClient.flush(THttpClient.java:349)
	at org.apache.thrift.TServiceClient.sendBase(TServiceClient.java:73)
	at org.apache.thrift.TServiceClient.sendBase(TServiceClient.java:62)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Client.send_getLicenseInfoFile(LicenseInfoService.java:162)
	at org.eclipse.sw360.datahandler.thrift.licenseinfo.LicenseInfoService$Client.getLicenseInfoFile(LicenseInfoService.java:150)
	at org.eclipse.sw360.portal.portlets.projects.ProjectPortlet.downloadLicenseInfo(ProjectPortlet.java:278)
	at org.eclipse.sw360.portal.portlets.projects.ProjectPortlet.serveResource(ProjectPortlet.java:175)
	at com.liferay.portlet.FilterChainImpl.doFilter(FilterChainImpl.java:122)
	at com.liferay.portlet.ScriptDataPortletFilter.doFilter(ScriptDataPortletFilter.java:82)
```

**Comment by mcjaeger** [comment]
merging with bug: https://github.com/siemens/sw360/issues/12

**Pull request merged**